File: labels.py
#!/usr/bin/python

import sys

file = str(sys.argv[1])
lines = open(file, "r")
lines = map(str.strip, lines)
lines = filter(None, lines)

labels = {}
l = 0
output = ""
for line in lines:
	if line.endswith(':'):
		labels[line[:-1]] = l
	else:
		output += line + '\n'
		l += 1

for label in labels.keys():
	output = output.replace(label, str(labels[label]))

print output

File: js-translate/lib/labels.js
module.exports = function (code) {
    var result = [];
    var currentLine = 0;
    var labels = {};
    for (var i = 0; i < code.length; i++) {
        var line = code[i];
        if (line[line.length - 1] === ':') {
            var label = line.slice(2, -1);
            labels[label] = currentLine;
            line = '; ' + label + ':';
        } else {
            line = '    ' + line;
            currentLine++;
        }
        result.push(line);
    }

    var cl = 0;
    for (var i = 0; i < result.length; i++) {
        for (var label in labels) {
            result[i] = result[i].replace('__' + label, labels[label]);
            result[i] = result[i].replace('__(line + 2)', cl + 2);
        }
        if (result[i][0] !== ';') {
            cl++;
        }
    }

    return result;
};
